scipy/optimize/_qap.py

```diff
@@ -0,0 +1,418 @@
+import numpy as np
+import math
+from . import linear_sum_assignment, minimize_scalar, OptimizeResult
+
+
+def quadratic_assignment(
+    cost_matrix,
+    dist_matrix,
+    seed=None,
+    maximize=False,
+    n_init=1,
+    init_method="barycenter",
+    max_iter=30,
+    shuffle_input=True,
+    eps=0.05,
+):
+    """
+    Solve the quadratic assignment problem
+
+    This class solves the Quadratic Assignment Problem (QAP) and the
+    Graph Matching Problem through an implementation of the Fast
+    Approximate QAP Algorithm (FAQ) (these two problems are the same up
+    to a sign change) [1]_.
+
+    This algorithm can be thought of as finding an alignment of the
+    vertices of two graphs which minimizes the number of induced edge
+    disagreements, or, in the case of weighted graphs, the sum of squared
+    differences of edge weight disagreements. The option to add seeds
+    (known vertex correspondence between some nodes) is also available
+    [2]_.
+
+    Note that the quadratic assignment problem is NP-hard, is not
+    known to be solvable in polynomial time, and is computationally
+    intractable. Therefore, the results given are approximations,
+    not guaranteed to be exact solutions.
+
```

Comment:
I think adding your mathematical problem here is great like `linprog` doc: https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.linprog.html